### ISSUE TITLE
[naming-convention] Unfit with ESlint

### ISSUE THREAD

**Issue by Mister-Hope** [typescript-eslint/typescript-eslint#2074]
See this demo: 

https://eslint.org/demo#eyJ0ZXh0IjoiY29uc3QgZW5oYW5jZUFwcCA9ICh7IFZ1ZSB9KSA9PiB7XG5cdFxufTtcblxuY29uc3QgZGVmaW5lID0gKCkgPT4gKHtcbiAgICBDT01NRU5UX09QVElPTlM6IDEsXG59KTtcblxuY29uc3QgUkdCQVBhdHRlcm4gPSAvcmdiYVxcKCguKyk/LCguKyk/LCguKyk/LCguKyk/XFwpL3U7XG5jb25zdCBSR0JQYXR0ZXJuID0gL3JnYlxcKCguKyk/LCguKyk/LCguKyk/XFwpL3U7XG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjYsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsiY2FtZWxjYXNlIjoyfSwiZW52Ijp7fX19

No errors are triggered by eslint.

But when I am using 

```
// the default config essentially does the same thing as ESLint's camelcase rule
const defaultOptions: Options = [
  {
    selector: 'default',
    format: ['camelCase'],
    leadingUnderscore: 'allow',
    trailingUnderscore: 'allow',
  },

  {
    selector: 'variable',
    format: ['camelCase', 'UPPER_CASE'],
    leadingUnderscore: 'allow',
    trailingUnderscore: 'allow',
  },

  {
    selector: 'typeLike',
    format: ['PascalCase'],
  },
];
```

from [this doc](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/naming-convention.md) as 

```yml

      "@typescript-eslint/naming-convention":
        - warn
        - selector: default
          format:
            - camelCase
          leadingUnderscore: allow
          trailingUnderscore: allow

        - selector: variable
          format:
            - camelCase
            - UPPER_CASE
          leadingUnderscore: allow
          trailingUnderscore: allow

        - selector: typeLike
          format:
            - PascalCase
```

I got errors:

```
  9:7  warning  Property name COMMENT_OPTIONS must match one of the following formats: camelCase  @typescript-eslint/naming-convention

  8:35  warning  Parameter name Vue must match one of the following formats: camelCase  @typescript-eslint/naming-convention
  8:35  warning  Property name Vue must match one of the following formats: camelCase   @typescript-eslint/naming-convention

  54:11  warning  Variable name RGBAPattern must match one of the following formats: camelCase, UPPER_CASE  @typescript-eslint/naming-convention
  55:11  warning  Variable name RGBPattern must match one of the following formats: camelCase, UPPER_CASE   @typescript-eslint/naming-convention
```

So I think there is something wrong with the default config or the rule itself

**Comment by bradzacher** [comment]
This rule is stricter than camelcase.
camelcase has been around for a long time so there's a lot of things it doesn't check because of breaking changes.

If you don't want to enforce naming for properties, you can configure the rule to ignore them.

**Comment by Mister-Hope** [comment]
Maybe you should change to docs to describe this?
